src/Algebra/Construct/Sub/Ring/Ideal.agda

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Ideals of a ring
+--
+-- Based on Nathan van Doorn (Taneb)'s original PR
+-- https://github.com/agda/agda-stdlib/pull/2855
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible #-}
+
+open import Algebra.Bundles using (Ring)
+
+-- As with `Algebra.Construct.Sub.Group.Normal`, this module
+-- relocates Taneb's original `Algebra.Ideal`
+-- to a level one *below* that of `Algebra.Construct.Sub.Ring`,
+-- notwithstanding that we have yet to define `Sub.XRing`s...
+
+module Algebra.Construct.Sub.Ring.Ideal {c ℓ} (R : Ring c ℓ) where
```

Comment:
Same comment about this file not containing a construction but rather a definition. And it isn't `Sub` either: an `Ideal` is defined as a property of a `Ring` R. This happens to use a `Sub-bimodule` as part of its definition, sure, but it is not a `Subring`. So I think both `Construct` and `Sub` don't belong in the name.

Reply:
Yeah, I wondered about this. 
Do we regard `NormalSubgroup`, `Ideal` etc. as 'special' kinds of `Sub`object? 
Or is the specialness that hey give rise to `Quotient`s?
I'd previously suggested to @Taneb a `Quotientable` hierarchy, and then `Quotient.X` takes an `X` and a `Quotientable.X` as parameters... but got pushback on that idea.

#2854 and #2855 put these things at the top of `Algebra`, and that doesn't seem right to me either.